### PR TITLE
Use embedded metrics-profiles in index subcmd

### PR DESCRIPTION
**Pull request by rsevilla87**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description

With this PR, the index subcmd will be able to use metrics profiles embedded in the `kube-burner-ocp` binary, ~~when the flag --embedded is passed~~

Using an embedded profile
```shell
$ kube-burner-ocp index --metrics-profile metrics.yml  --check-health=false
time="2024-11-07 17:02:32" level=info msg="📁 Creating local indexer: indexer-0" file="metrics.go:67"
time="2024-11-07 17:02:32" level=info msg="👽 Initializing prometheus client with URL: https://prometheus-k8s-openshift-monitoring.apps.kube-burner-ocp.perfscale.devcluster.openshift.com" file="prometheus.go:47"
time="2024-11-07 17:02:32" level=info msg="🔍 Endpoint: https://prometheus-k8s-openshift-monitoring.apps.kube-burner-ocp.perfscale.devcluster.openshift.com; profile: config/metrics.yml start: 2024-11-07T16:02:26+01:00 end: 2024-11-07T17:12:26+01:00; job: kube-burner-ocp-indexing" file="prometheus.go:70"
```

Using a local file as profile:
```shell
$ ./bin/amd64/kube-burner-ocp index --metrics-profile custom-metrics.yml  --check-health=false
time="2024-11-07 17:04:57" level=info msg="📁 Creating local indexer: indexer-0" file="metrics.go:67"
time="2024-11-07 17:04:57" level=info msg="👽 Initializing prometheus client with URL: https://prometheus-k8s-openshift-monitoring.apps.kube-burner-ocp.perfscale.devcluster.openshift.com" file="prometheus.go:47"
time="2024-11-07 17:04:58" level=info msg="Embedded config doesn't contain metrics profile config/custom-metrics.yml. Falling back to original path" file="prometheus.go:139"
time="2024-11-07 17:04:58" level=info msg="🔍 Endpoint: https://prometheus-k8s-openshift-monitoring.apps.kube-burner-ocp.perfscale.devcluster.openshift.com; profile: custom-metrics.yml start: 2024-11-07T16:04:52+01:00 end: 2024-11-07T17:14:52+01:00; job: kube-burner-ocp-indexing" file="prometheus.go:70"
``


## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: cmd/ocp.go [hunk 1]
```diff
@@ -32,8 +32,6 @@ import (
 //go:embed config/*
 var ocpConfig embed.FS
 
-const configDir = "config"
```
Comment (vishnuchalla) [inline cmd/ocp.go:35]: I think its okay to keep this constant here itself. It is anyways not being used in other files.

File: index.go [hunk 2]
```diff
@@ -102,11 +104,16 @@ func NewIndex(metricsEndpoint *string, ocpMetaAgent *ocpmetadata.Metadata) *cobr
 				metadata[k] = v
 			}
 			workloads.ConfigSpec.MetricsEndpoints = append(workloads.ConfigSpec.MetricsEndpoints, indexer)
+			if embedded {
```
Comment (vishnuchalla) [inline index.go:107]: If embedded is not true `ocpConfig` function parameter won't even be used. I was just thinking if we want to provide this embedded feature in `ocp.go` and make it available for other workloads too?

File: cmd/ocp.go (hunk 1 above)
Comment (rsevilla87) [reply cmd/ocp.go:35]: I moved it to common.go since it's referenced in index.go

File: index.go (hunk 2 above)
Comment (rsevilla87) [reply index.go]: I have refactored part of the pr to remove this flag, PTAL again

**Review by vishnuchalla — APPROVED** [review]
lgtm

**Pull request merged**